### PR TITLE
[#41] Migrate to Spring Data

### PR DESCRIPTION
**Pull request by Sheptytskyid**

### CODE REVIEW

File: src/main/java/net/greatstart/configs/SecurityConfig.java [hunk 1]
```diff
@@ -12,19 +12,24 @@
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
+import javax.sql.DataSource;
+
 @Configuration
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
     private UserService userService;
+    private DataSource dataSource;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/configs/SecurityConfig.java]: MAJOR Remove this unused "dataSource" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by holkwen** [comment]
У меня новые проекты перестали добавляться.

**Comment by holkwen** [comment]
Лечится добавлением в pom.xml:
        &lt;dependency&gt;
            &lt;groupId&gt;org.hibernate&lt;/groupId&gt;
            &lt;artifactId&gt;hibernate-java8&lt;/artifactId&gt;
            &lt;version&gt;${hibernate.version}&lt;/version&gt;
        &lt;/dependency&gt;

File: pom.xml [hunk 2]
```diff
@@ -130,6 +142,12 @@
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
                 <version>0.7.8</version>
+                <configuration>
```
Comment (zozich) [inline pom.xml:133]: This is not needed. Your `model` and `dto` classes should be tested withing testing other layers

File: pom.xml [hunk 3]
```diff
@@ -87,10 +82,27 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.powermock</groupId>
```
Comment (zozich) [inline pom.xml]: Remove if not needed

File: src/main/java/net/greatstart/dao/ProjectDao.java [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+import org.springframework.data.repository.CrudRepository;
 
 import java.util.List;
 
-public interface ProjectDao extends GenericDao<Project> {
-    List<Project> getNRecords(int numberOfProjects);
+public interface ProjectDao extends CrudRepository<Project, Long> {
 
-    List<Project> getByUserId(long id);
+    List<Project> findAll(Pageable numberOfRecords);
```
Comment (zozich) [inline src/main/java/net/greatstart/dao/ProjectDao.java]: Don't you have it by default?

File: src/main/java/net/greatstart/dto/DtoUser.java [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
 
 import lombok.Data;
+import org.springframework.stereotype.Component;
 
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Pattern;
 import javax.validation.constraints.Size;
 
 @Data
+@Component
```
Comment (zozich) [inline src/main/java/net/greatstart/dto/DtoUser.java]: Why?

File: src/main/java/net/greatstart/dto/DtoUserProfile.java [hunk 6]
```diff
@@ -1,11 +1,13 @@
 package net.greatstart.dto;
 
 import lombok.Data;
+import org.springframework.stereotype.Component;
 
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Size;
 
 @Data
+@Component
```
Comment (zozich) [inline src/main/java/net/greatstart/dto/DtoUserProfile.java]: Why? Remove everywhere

File: src/main/java/net/greatstart/controllers/EventController.java [hunk 7]
```diff
@@ -17,16 +17,12 @@ public EventController(EventService eventService) {
         this.eventService = eventService;
     }
 
-    public boolean createEvent(Event event) {
-        return eventService.createEvent(event);
+    public void saveEvent(Event event) {
+        eventService.saveEvent(event);
```
Comment (zozich) [inline src/main/java/net/greatstart/controllers/EventController.java]: save should return saved entity as done by default by Spring Data

**Review by zozich — CHANGES_REQUESTED** [review]
Please amend

**Review by zozich — APPROVED** [review]
Okay now

**Pull request merged**